apps/qtmozembed/declarativewebpage.cpp

```diff
@@ -398,35 +372,8 @@ void DeclarativeWebPage::thumbnailReady()
 
 void DeclarativeWebPage::updateViewMargins()
 {
-    qCDebug(lcCoreLog) << "WebPage: update view margins, foreground:" << (m_container && !m_container->foreground()) << "throttling:" << (m_marginChangeThrottleTimer > 0) << "moving:" << moving();
-
-    // Don't update margins while panning, flicking, pinching, vkb is already open, or
-    // margin update is ongoing (throttling).
-    if ((m_container && !m_container->foreground()) || m_marginChangeThrottleTimer > 0 ||
-            moving() || m_virtualKeyboardMargin > 0) {
-        return;
-    }
-
-    resetViewMargins();
-}
-
-void DeclarativeWebPage::resetViewMargins()
-{
-    qCDebug(lcCoreLog) << "WebPage: reset view margins, fullscreen:" << m_fullscreen << "content height:" << contentHeight();
-
-    // Reset margins always when fullscreen mode is enabled.
     QMargins margins;
-    if (!m_fullscreen) {
-        qreal threshold = qMax(m_fullScreenHeight * 1.5f, (m_fullScreenHeight + (m_toolbarHeight*2)));
-        if (contentHeight() < threshold) {
-            margins.setBottom(m_toolbarHeight);
-        }
-    }
-
-    // Some content needed so that it makes sense to throttle content height changes.
-    if (contentHeight() > 0) {
-        m_marginChangeThrottleTimer = startTimer(200);
-    }
+    margins.setBottom(qMax(virtualKeyboardHeight(), (int)toolbarHeight()));
 
     qCDebug(lcCoreLog) << "WebPage: set margins:" << margins;
```

Comment:
With the animated toolbar change, this generates a stream of debug output. It's only when `lcCoreLog` logging is enabled, but maybe it's better to remove it anyway? What do you think?